core/src/main/scala/org/apache/spark/sql/delta/commands/WriteIntoDelta.scala

```diff
@@ -149,6 +154,18 @@ case class WriteIntoDelta(
       }
     }
 
+    val useDynamicPartitionOverwriteMode = {
+      val useDynamic = txn.metadata.partitionColumns.nonEmpty &&
+        options.isDynamicPartitionOverwriteMode
+      replaceWhere.foreach { replace =>
+        if (useDynamic) {
+          throw new AnalysisException("'replaceWhere' cannot be used when " +
+            "'partitionOverwriteMode' is set to 'DYNAMIC'")
```

Comment:
use the option name PARTITION_OVERWRITE_MODE_OPTION?

Reply:
fixed